crate/findex/src/adt.rs

```diff
@@ -32,21 +50,31 @@ pub trait IndexADT<Keyword: Send + Hash, Value: Send + Hash> {
     ) -> impl Send + Future<Output = Result<(), Self::Error>>;
 }
 
-pub trait VectorADT: Send {
-    /// Vectors are homogeneous.
-    type Value: Send;
+/// BatcherSSEADT (Batched Searchable Symmetric Encryption Abstract Data Type)
+///
+/// This trait defines batch operations for encrypted searchable indices. It extends
+/// the functionality of the standard `IndexADT` by providing methods that operate on
+/// multiple keywords or entries simultaneously to cut network's overhead and improve performance.
+pub trait BatchedIndexADT<Keyword, Value> {
+    type Error: std::error::Error;
 
-    /// Vector error.
-    type Error: Send + std::error::Error;
+    /// Search the index for the values bound to the given keywords.
+    fn batch_search(
+        &self,
+        keywords: Vec<&Keyword>, // n  --> n fois barch read --> n+1
```

Comment:
Please read your diffs! This comment is ugly and does not mean anything!
```suggestion
        keywords: Vec<&Keyword>,
```

Reply:
missed it, done